Rearchitecture of Dashboard features

maznnwell opened the pull request:
Summary of changes, manly found in the lib/rearchitecture folder:

- created a new entry point of the app: main_dev.dart that skips the splash screen; to use this entry point, you can import the run configuration from run_configurations/dev.run.xml
- implemented a feature based architecture, using Bloc as state management solution; all the new, or refactored code, will be found in the lib/rearchitecture folder; in a feature based architecture, we create a folder and add all related code to it inside that folder
- refactored code and created features to be used in the "Dashboard" tab - with the exception of "Plasma" and "Latest Transactions" widgets that I considered to be part of another mother tab
- used HydratedCubits, instead of the inhouse blocs; cubits are lightweight blocs that handle one single job; HydratedCubits cache their last state and restores it
- I've added the https://pub.dev/packages/very_good_analysis package for linter rules; it's far more stricter than the previous solution; I haven't solved all the warnings and hints in the app, because that would make my PR too wide in terms of scope
- I've created a new version of CardScaffold, one that does not listens to a stream anymore; this new widget also uses the new app themes that are more closer to the default Material Design used by Flutter, especially when it comes to the text styles: I preferred to used the original ones, instead of overwriting them
- fixed the switch from the back of the new cards - when active, the thumb was disappearing
- fixed the feature of hiding, unhiding the front widget of card (the password field was not being dismissed properly, and the switch was showing that the discreet mode was not active, but front widget was still hidden)
- started using localized strings, so it will be easier to translate the app later
- added unit tests for the cubits used

The high amount of changes in the folders outside lib/rearchitecture are due to running dart fix --apply, that fixed warnings from the new linter rules. One new rule that I wanted to use specifically, and that generated a lot of changes, is the one that requests strict types when declaring a local variable. I've added comments for this choice in the analysis_options.yaml file

KingGorrin commented on windows/CMakeLists.txt:
```diff
@@ -94,39 +106,3 @@ install(DIRECTORY "${PROJECT_BUILD_DIR}/${FLUTTER_ASSET_DIR_NAME}"
-
-find_program(POWERSHELL_PATH NAMES powershell)
-add_custom_target("git_metadata" ALL
-  COMMAND ${POWERSHELL_PATH} -ExecutionPolicy Bypass "${CMAKE_HOME_DIRECTORY}/CreateGitMetadata.ps1")
```
Changes from line 97 to 132 causes the build not to copy the necessary binaries library files and the metadata not to be populated with the git information.

KingGorrin replied:
Just so you know, this happens on all three platforms. Changes are made to the `CMakeLists.txt` and `project.pbxproj` files. I'm not entirely convinced this is the recommended way of doing it.

KingGorrin commented on windows/runner/Runner.rc:
```diff
@@ -89,11 +89,11 @@ BEGIN
         BEGIN
-            VALUE "CompanyName", "network.zenon.syrius" "\0"
-            VALUE "FileDescription", "s y r i u s (" VERSION_AS_STRING ")" "\0"
+            VALUE "CompanyName", "com.example" "\0"
```
Why is the company name changed?

KingGorrin commented on windows/runner/Runner.rc:
```diff
@@ -89,11 +89,11 @@ BEGIN
             VALUE "FileVersion", VERSION_AS_STRING "\0"
             VALUE "InternalName", "syrius" "\0"
-            VALUE "LegalCopyright", "(C) 2024 Zenon Network" "\0"
+            VALUE "LegalCopyright", "Copyright (C) 2024 com.example. All rights reserved." "\0"
```
Why is the LegalCopyright statement changed?

KingGorrin commented on windows/runner/resources/app_icon.ico:
This overwrites the new syrius icon with the default flutter icon.

KingGorrin commented on windows/runner/win32_window.cpp:
I believe this file has been manually changed to include support for App Links. The changes remove the App Links support.

KingGorrin reviewed (commented):
- Icon has been reset to default flutter icon. See screenshot.
- Header font size of new scaffold card is bigger causing a discrepancy between old and new cards. See screenshot.
- The Pillar Rewards values in the vertical column do not fit. See screenshot.
- Metadata information is not being build. See comments.
- App Links support has been removed.

![image](https://github.com/user-attachments/assets/a5af4418-2227-45a4-9979-2f27725d1563)

![image](https://github.com/user-attachments/assets/29e3cbc6-8069-4f98-b5db-692324ca2c1c)

![image](https://github.com/user-attachments/assets/3eea56d8-45ca-4bf0-9d2f-7aa51cdd1ddf)

KingGorrin commented on pubspec.yaml:
```diff
@@ -43,8 +44,8 @@ dependencies:
-      url: https://github.com/zenon-network/znn_sdk_dart.git
-      ref: v0.0.7
+      url: https://github.com/maznnwell/znn_sdk_dart.git
+      ref: 1fa766fc342360dfed93a2fce247d69674cb7272
```
You can update to v0.0.8

KingGorrin commented on pubspec.yaml:
```diff
@@ -69,18 +70,38 @@ dependencies:
+  win32: ^5.9.0
+
+
+dependency_overrides:
```
Overrides can be removed

KingGorrin commented:
Not sure why exactly, but the default app icon is only used in the task bar.

![image](https://github.com/user-attachments/assets/d7fece33-3d6e-49fa-a80c-6df83776ee54)

maznnwell commented:
> Not sure why exactly, but the default app icon is only used in the task bar. The app icon and system tray are restored.
> 
> ![image](https://private-user-images.githubusercontent.com/13164589/392584914-d7fece33-3d6e-49fa-a80c-6df83776ee54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MDY2OTUsIm5iZiI6MTczMzQwNjM5NSwicGF0aCI6Ii8xMzE2NDU4OS8zOTI1ODQ5MTQtZDdmZWNlMzMtM2Q2ZS00OWZhLWE4MGMtNmRmODM3NzZlZTU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDEzNDYzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2YjgxNTkxMDc1MDQxZmNmNDM4YmU3NzY4ODUyZTM5NGQyOGQzMzNkMDk5NTlkZjUxMGMzY2NlNmQ2MTJiNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ulzvB8IEhuYSr4UY2av56At4aYgdCY8B37G-nbVVC1E)

I build the app using `flutter build windows --release` and all the three icons are the Syrius icon.

I also fixed the build on macOS and Linux. There was a problem with building some paths for some dll or .so files.

maznnwell commented:
> * Icon has been reset to default flutter icon. See screenshot.
> * Header font size of new scaffold card is bigger causing a discrepancy between old and new cards. See screenshot.
> * The Pillar Rewards values in the vertical column do not fit. See screenshot.
> * Metadata information is not being build. See comments.
> * App Links support has been removed.
> 
> ![image](https://private-user-images.githubusercontent.com/13164589/389204803-a5af4418-2227-45a4-9979-2f27725d1563.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MDY2OTYsIm5iZiI6MTczMzQwNjM5NiwicGF0aCI6Ii8xMzE2NDU4OS8zODkyMDQ4MDMtYTVhZjQ0MTgtMjIyNy00NWE0LTk5NzktMmYyNzcyNWQxNTYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDEzNDYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4Y2ZmNzA0MzI3MGNmZWRlNzdhZWQzNTNjMmZmYmUxNGQ2ZmMyNTJlMDAzYjgwODlmYTA4YzYyM2ZhYTNhZDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.O4qsNzAIKyICrOrufsVqsuxCzpfeUMjkPfoLGJilXsw)
> 
> ![image](https://private-user-images.githubusercontent.com/13164589/389204781-29e3cbc6-8069-4f98-b5db-692324ca2c1c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MDY2OTYsIm5iZiI6MTczMzQwNjM5NiwicGF0aCI6Ii8xMzE2NDU4OS8zODkyMDQ3ODEtMjllM2NiYzYtODA2OS00Zjk4LWI1ZGItNjkyMzI0Y2EyYzFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDEzNDYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhMmNjYzlkYTE2ZTkzZWRjOTZiYzYzNGU4NzY0YjJkM2ExYzMzY2MzNTE5NGQ2YzE0ZTdjYjJkOThlYTc1MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3OFmd6wjHSvIIjtj5qHakBK1zF0kkJ2tf1hbP_9chOI)
> 
> ![image](https://private-user-images.githubusercontent.com/13164589/389204733-3eea56d8-45ca-4bf0-9d2f-7aa51cdd1ddf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MDY2OTYsIm5iZiI6MTczMzQwNjM5NiwicGF0aCI6Ii8xMzE2NDU4OS8zODkyMDQ3MzMtM2VlYTU2ZDgtNDVjYS00YmYwLTlkMmYtN2FhNTFjZGQxZGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDEzNDYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1YmY5NjE2ZmMyMzIyMzYxYTQ2MzZhYmI1NGVlYjM5MTczNDg2YWY5MjQzN2M3YjRmNmQ4NTUxNGZhMzYxNTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1R-VeRVcTK5oMxWG5km5E-jUGiSJrerpPSoag2JSEiM)

I know about the misalignment of the two cards. Its intended because I haven't refactored the Plasma card from the Dashboard tab, I want to do it when I refactor the whole plasma tab; after I finish the Plasma tab, everything will be even on Dashboard as well. But I can make it sooner if it's a deal breaker.

I changed how the Y values are generated on charts. And added more space for the Y values.

I restored the generation of metadata.

Enabled full app_links support: the Windows app can now be opened by an URL using the scheme syrius://. I can go to the bridge and open the app from it in order to test it in combination with the WC feature.

KingGorrin commented:
> I know about the misalignment of the two cards. Its intended because I haven't refactored the Plasma card from the Dashboard tab, I want to do it when I refactor the whole plasma tab; after I finish the Plasma tab, everything will be even on Dashboard as well. But I can make it sooner if it's a deal breaker.

You're right, this will be solved once all widgets have been refactored. Let's leave this as it is for now.

KingGorrin commented:
> > Not sure why exactly, but the default app icon is only used in the task bar. The app icon and system tray are restored.
> > ![image](https://private-user-images.githubusercontent.com/13164589/392584914-d7fece33-3d6e-49fa-a80c-6df83776ee54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MDY2OTUsIm5iZiI6MTczMzQwNjM5NSwicGF0aCI6Ii8xMzE2NDU4OS8zOTI1ODQ5MTQtZDdmZWNlMzMtM2Q2ZS00OWZhLWE4MGMtNmRmODM3NzZlZTU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDEzNDYzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2YjgxNTkxMDc1MDQxZmNmNDM4YmU3NzY4ODUyZTM5NGQyOGQzMzNkMDk5NTlkZjUxMGMzY2NlNmQ2MTJiNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ulzvB8IEhuYSr4UY2av56At4aYgdCY8B37G-nbVVC1E)
> 
> I build the app using `flutter build windows --release` and all the three icons are the Syrius icon.
> 
> I also fixed the build on macOS and Linux. There was a problem with building some paths for some dll or .so files.

Turns out to be a caching issue on my side. It's working now.

maznnwell commented:
I want to leave a comment that explains the part of code that finds the path for some .dll and .so files and outputs a copy of those in the application folder. This process happens for each desktop build: Windows, Linux and macOS. The app needs some libraries to function, like the libpow library. The app looks for those files in multiple locations, such as the root folder of the application. But they are original found only in the SDK, as is the case of the libpow library.

So, for each desktop build, there is a some code that looks up the path of the znn_sdk_dart library in the local cache of the machine. And sincer we already know where the libpow dll is located inside the znn_sdk_dart, we can append the two paths and install the dll inside the application folder.

The issue is that originally, the znn_sdk_dart was searched for inside the dart_tool/package_config.json file. It looks up the path by going through the file one line at a time. The catch is that the package_config.json in the Dashboard-PR is generated now as one long line. I don't know what caused the change from generating a multi-line file to a single line file.

So I decided to look for the paths inside the dart_tool/package_config_subset that seemed to be generated in a consistent way.

maznnwell commented on lib/utils/metadata.dart:
```diff
@@ -1,5 +1,5 @@
-const String gitBranchName = '';
```
Are there any changes still needed here?

KingGorrin replied:
Yes, you have to restore the file and remove your local git information.

maznnwell replied:
Done.

KingGorrin reviewed (approved):
Only the `metadata.dart` file needs to be restored.

KingGorrin commented:
Found an issue with the rendering of the AZ project/phase list. A part of the project and phase list is blanked out and blocks further scrolling. Seems like the widgets crash. See screenshots.

![image](https://github.com/user-attachments/assets/368b411b-5eeb-4f56-9f93-8ff434add587)
![image](https://github.com/user-attachments/assets/4ac153a8-2a0d-4559-aabe-f34eab6a36f6)

maznnwell commented:
> Found an issue with the rendering of the AZ project/phase list. A part of the project and phase list is blanked out and blocks further scrolling. Seems like the widgets crash. See screenshots.
> 
> ![image](https://private-user-images.githubusercontent.com/13164589/393594161-368b411b-5eeb-4f56-9f93-8ff434add587.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3ODAxMjksIm5iZiI6MTczMzc3OTgyOSwicGF0aCI6Ii8xMzE2NDU4OS8zOTM1OTQxNjEtMzY4YjQxMWItNWVlYi00ZjU2LTlmOTMtOGZmNDM0YWRkNTg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDIxMzAyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzNWZjY2JkYjE0Mzg1NTA3MGY2ZmY2MjAwYmUwNzdiMzcwOWQ1MzA5MWJiMTdiYWViZmQwZjFjOGMyZDVkOGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.99bWkEzrMfgMRf_sy8Nblw3Khnz2-oUBWfvwNaMnZWU) ![image](https://private-user-images.githubusercontent.com/13164589/393594168-4ac153a8-2a0d-4559-aabe-f34eab6a36f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3ODAxMjksIm5iZiI6MTczMzc3OTgyOSwicGF0aCI6Ii8xMzE2NDU4OS8zOTM1OTQxNjgtNGFjMTUzYTgtMmEwZC00NTU5LWFhYmUtZjM0ZWFiNmEzNmY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDIxMzAyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxYWJiYzQ1Mzc5NjVkOThjYzMwMTA1ZDYxNWE0ZGQ2YzhiMGJlY2RiMDg2YWZkNTMwNmZiYTQwMGUyNzM5NjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MQ2GcqPar7pm_GGyfmuyA9wtHCIXBKiKp1Nl1AC0dCQ)

Done.
